pkg/controller/node/ipam/timeout.go

```diff
@@ -0,0 +1,67 @@
+/*
```

Comment:
Not sure if you already looked at this and found it not be usable for your cases, however there is this:
https://github.com/kubernetes/apimachinery/blob/master/pkg/util/wait/wait.go#L192